common/client_test.go

```diff
@@ -0,0 +1,332 @@
+			})
+
+			It("should use a http client with tls verification disabled", func() {
+				_, _ = actualRodeClient.GetPolicy(context.Background(), &pb.GetPolicyRequest{})
+
+				mockTransport, ok := insecureOauthHttpClient.Transport.(*httpmock.MockTransport)
+				Expect(ok).To(BeTrue())
+
+				httpmock.GetCallCountInfo()
+				Expect(mockTransport.GetCallCountInfo()).ToNot(HaveLen(0))
+			})
+		})
+
+		When("the client ID is not specified", func() {
+			BeforeEach(func() {
+				expectedConfig.JWTAuth.ClientID = ""
+			})
+
+			It("should return an error", func() {
+				Expect(actualRodeClient).To(BeNil())
+				Expect(actualError).To(HaveOccurred())
+			})
+		})
+
+		When("the client secret is not specified", func() {
+			BeforeEach(func() {
+				expectedConfig.JWTAuth.ClientSecret = ""
+			})
+
+			It("should return an error", func() {
+				Expect(actualRodeClient).To(BeNil())
+				Expect(actualError).To(HaveOccurred())
+			})
+		})
+
+		When("the token URL is not specified", func() {
+			BeforeEach(func() {
+				expectedConfig.JWTAuth.TokenURL = ""
+			})
+
+			It("should return an error", func() {
+				Expect(actualRodeClient).To(BeNil())
+				Expect(actualError).To(HaveOccurred())
+			})
+		})
+	})
+
+	When("basic auth is configured", func() {
+		var (
+			expectedUsername string
+			expectedPassword string
+		)
+
+		BeforeEach(func() {
+			expectedUsername = fake.LetterN(10)
+			expectedPassword = fake.LetterN(10)
+
+			expectedConfig.BasicAuth = &BasicAuthConfig{
+				Username: expectedUsername,
+				Password: expectedPassword,
+			}
+		})
+
+		It("should return a rode client", func() {
+			Expect(actualRodeClient).ToNot(BeNil())
+			Expect(actualError).ToNot(HaveOccurred())
+		})
+
+		It("should send a basic authentication header with each request", func() {
+			_, _ = actualRodeClient.GetPolicy(context.Background(), &pb.GetPolicyRequest{})
+
+			Expect(actualAuthenticationHeader).ToNot(BeEmpty())
+
+			parts := strings.Split(actualAuthenticationHeader, " ")
+
+			Expect(parts[0]).To(Equal("Basic"))
+
+			data, err := base64.StdEncoding.DecodeString(parts[1])
+			Expect(err).ToNot(HaveOccurred())
+
+			dataParts := strings.Split(string(data), ":")
+			Expect(dataParts[0]).To(Equal(expectedUsername))
+			Expect(dataParts[1]).To(Equal(expectedPassword))
+		})
+
+		When("the username is missing", func() {
+			BeforeEach(func() {
+				expectedConfig.BasicAuth.Username = ""
+			})
+
+			It("should return an error", func() {
+				Expect(actualRodeClient).To(BeNil())
+				Expect(actualError).To(HaveOccurred())
+			})
+		})
+
+		When("the password is missing", func() {
+			BeforeEach(func() {
+				expectedConfig.BasicAuth.Password = ""
+			})
+
+			It("should return an error", func() {
+				Expect(actualRodeClient).To(BeNil())
+				Expect(actualError).To(HaveOccurred())
+			})
+		})
+	})
+
+	// TODO: figure out how to make this not take 10 seconds
+	XWhen("connecting to the server fails", func() {
```

Comment:
AFAIK we don't have any other skipped tests in the suites, so I think I'd rather there be an issue to follow up on than having this skipped. What do you think?

Reply:
I could also decrease the timeout, but it was really frustrating trying to run tests after every change and have the entire suite take 10 seconds because of this.  If there was a way to enable it in CI but disable it locally I'd prefer that.

Reply:
Maybe we just always run the test and it short circuits if `os.Getenv("CI")` is empty? It should be set to true in most CI platforms

Reply:
I ended up pulling the context duration out into a variable that the test can modify to decrease the timeout